pulp/tests/test_pulp.py

```diff
@@ -787,7 +787,16 @@ def test_timeLimit(self):
         # CHOCO has issues when given a time limit
         if self.solver.name != 'PULP_CHOCO_CMD':
             pulpTestCheck(prob, self.solver, [const.LpStatusOptimal], {x: 4, y: -1, z: 6, w: 0})
-
+    
+    def test_assignStatus(self):
```

Comment:
Most examples already test that `const.LpStatus` is filled correctly but they do not check `const.LpSolution`. Check the function just below `pulpTestCheck`. Maybe we can make that function test that for solution status. Also, it would be interesting to add tests were we add an incorrect status and raise the error or warning.

Reply:
I didn't notice that `const.LpStatus` has been tested, cause there is no directly `assignStatus()` test. For this PR, I thought invalid status check is a good idea.
The check for the validity of solution status would be easy, but the verification of solution status will be a massive modification.(Since it requires to know all the solution status of examples.)
I'll go for invalid status test first!